File: C++/2657FindThePrefixCommonArrayOfTwoArrays.cpp
//
//  2657FindThePrefixCommonArrayOfTwoArrays.cpp
//  Testing C++
//
//  Created by Cem Beyenal on 1/14/25.
//
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        vector<int> ans;
        unordered_set<int> mp;
        for (int i = 0; i < A.size(); i++) {
            mp.insert(A[i]);
            mp.insert(B[i]);
            ans.push_back((i + 1) * 2 - mp.size());
        }
        return ans;
    }
};

File: Swift/2657FindThePrefixCommonArrayOfTwoArrays.swift
class Solution {
    func findThePrefixCommonArray(_ A: [Int], _ B: [Int]) -> [Int] {
        var aMap = [Int: Int](), res = [Int](), count = 0
        for i in 0..<A.count {
            aMap[A[i], default: 0] += 1 
            if aMap[A[i]] == 2 {
                count += 1
            }
            aMap[B[i], default: 0] += 1 
            if aMap[B[i]] == 2 {
                count += 1
            }

            res.append(count)
            print(aMap)
        }



        return res
    }
}
